app/controllers/api/v2/parameters_controller.rb

```diff
@@ -34,7 +34,11 @@ class ParametersController < V2::BaseController
 
       def index
         base = nested_obj.send(parameters_method).authorized(current_permission)
-        @parameters = base.search_for(*search_options).paginate(paginate_options)
+        @parameters = if paginate_options[:per_page] == 'all'
+                        base.search_for(*search_options)
+                      else
+                        base.search_for(*search_options).paginate(paginate_options)
+                      end
```

Comment:
Elsewhere we use this pattern (or some variation on it):
```suggestion
        @parameters = base.search_for(*search_options)
        @parameters = @parameters.paginate(paginate_options) if paginate_options[:per_page] != 'all'
```

Would this incur another query, or what's the reason you didn't choose that?

Reply:
Good idea. Checked regarding extra queries - both solutions have the same count.